From a74ab71aaa47c7ecf7882c69a81524e90a138f02 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Fri, 21 Feb 2025 15:53:28 +0100
Subject: [PATCH 1628/1695] drivers: stpmic1: update the bindings

Update the bindings now that the interrupt framework is in use.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c36c141855c712075519491e2a923a65d2acc672 drivers: stpmic1: update the bindings]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Ib6be171126fb8028754a3137c9373793ddbe3325
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/441059
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 documentation/devicetree/bindings/mfd/st,stpmic1.yaml | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/documentation/devicetree/bindings/mfd/st,stpmic1.yaml b/documentation/devicetree/bindings/mfd/st,stpmic1.yaml
index 5c2b71854..e6de63119 100644
--- a/documentation/devicetree/bindings/mfd/st,stpmic1.yaml
+++ b/documentation/devicetree/bindings/mfd/st,stpmic1.yaml
@@ -276,12 +276,6 @@ properties:
 
     additionalProperties: false
 
-  st,wakeup-pin-number:
-    description: Pin used as wake-up source when for MP13
-    $ref: /schemas/types.yaml#/definitions/uint32
-    minItems: 1
-    maxItems: 1
-
   st,pmic-it-id:
     description: PMIC interruptions id to forward to the non-secure world
       in the form of OPTEE notifications described by st,notif-it-id.
@@ -351,13 +345,11 @@ examples:
       pmic@33 {
         compatible = "st,stpmic1";
         reg = <0x33>;
-        interrupt-parent = <&gpioa>;
-        interrupts = <0 2>;
 
         interrupt-controller;
         #interrupt-cells = <2>;
 
-        st,wakeup-pin-number = <1>;
+        interrupts-extended = <&exti 55 IRQ_TYPE_EDGE_FALLING>;
         st,pmic-it-id = <IT_PONKEY_F IT_PONKEY_R>;
         st,notif-it-id = <0 2>;
 
-- 
2.47.2

